packages/@aws-cdk/aws-eks-v2-alpha/lib/kubectl-handler/helm/__init__.py

```diff
@@ -0,0 +1,203 @@
+    public_ecr_pattern = 'oci://(?P<registry>public\.ecr\.aws)*'
+
+    private_registry = re.match(private_ecr_pattern, repository).groupdict()
+    public_registry = re.match(public_ecr_pattern, repository).groupdict()
+
+    if private_registry['registry'] is not None:
+        logger.info("Found AWS private repository")
+        cmnd = [
+            f"aws ecr get-login-password --region {private_registry['region']} | " \
+            f"helm registry login --username AWS --password-stdin {private_registry['registry']}; helm pull {repository} --version {version} --untar"
+            ]
+    elif public_registry['registry'] is not None:
+        logger.info("Found AWS public repository, will use default region as deployment")
+        region = os.environ.get('AWS_REGION', 'us-east-1')
+
+        if is_ecr_public_available(region):
+            cmnd = [
+                f"aws ecr-public get-login-password --region us-east-1 | " \
+                f"helm registry login --username AWS --password-stdin {public_registry['registry']}; helm pull {repository} --version {version} --untar"
+                ]
+        else:
+            # `aws ecr-public get-login-password` and `helm registry login` not required as ecr public is not available in current region
+            # see https://helm.sh/docs/helm/helm_registry_login/
+            cmnd = [f"helm pull {repository} --version {version} --untar"]
+    else:
+        logger.error("OCI repository format not recognized, falling back to helm pull")
+        cmnd = [f"helm pull {repository} --version {version} --untar"]
+
+    return cmnd
+
+
+def get_chart_from_oci(tmpdir, repository = None, version = None):
+
+    cmnd = get_oci_cmd(repository, version)
+
+    maxAttempts = 3
+    retry = maxAttempts
+    while retry > 0:
+        try:
+            logger.info(cmnd)
+            output = subprocess.check_output(cmnd, stderr=subprocess.STDOUT, cwd=tmpdir, shell=True)
+            logger.info(output)
+
+            # effectively returns "$tmpDir/$lastPartOfOCIUrl", because this is how helm pull saves OCI artifact.
+            # Eg. if we have oci://9999999999.dkr.ecr.us-east-1.amazonaws.com/foo/bar/pet-service repository, helm saves artifact under $tmpDir/pet-service
+            return os.path.join(tmpdir, repository.rpartition('/')[-1])
+        except subprocess.CalledProcessError as exc:
+            output = exc.output
+            if b'Broken pipe' in output:
+                retry = retry - 1
+                logger.info("Broken pipe, retries left: %s" % retry)
+            else:
+                raise Exception(output)
+    raise Exception(f'Operation failed after {maxAttempts} attempts: {output}')
+
+
+def helm(verb, release, chart = None, repo = None, file = None, namespace = None, version = None, wait = False, timeout = None, create_namespace = None, skip_crds = False, atomic = False):
+    import subprocess
+
+    cmnd = ['helm', verb, release]
+    if not chart is None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Use `is not` operator rather than `not ... is` because it is more readable and makes your code easier to understand.

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=Use%20is%20not)

Similar issue at line numbers 171, 173, 175, 177, and 183.